# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/frontend/FunctionEmitter.cpp
# Commit: d96f98f974e0
# Full Hash: d96f98f974e087224e0a1ef9be67e2fc63397d8b
# Author: Ashley Hauck <khyperia@mozilla.com>
# Date: 2019-03-21 04:34:45
# Regressor Bug: 1535166
# File Overlap Count: 1
# Description:
#   Bug 1535166 - Implement computed field names. r=jorendorff
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D23408
# ==============================================================================

diff -r c786b72a9d8e -r d96f98f974e0 js/src/frontend/FunctionEmitter.cpp
--- a/js/src/frontend/FunctionEmitter.cpp	Wed Mar 20 15:50:30 2019 +0000
+++ b/js/src/frontend/FunctionEmitter.cpp	Wed Mar 20 17:26:01 2019 +0000
@@ -627,7 +627,10 @@
       return false;
     }
 
-    if (!bce_->emit1(JSOP_CALLELEM)) {
+    // Don't use CALLELEM here, because the receiver of the call != the receiver
+    // of this getelem. (Specifically, the call receiver is `this`, and the
+    // receiver of this getelem is `.initializers`)
+    if (!bce_->emit1(JSOP_GETELEM)) {
       //            [stack] ARRAY? FUNC
       return false;
     }